taskchampion/lib/src/server.rs

```diff
@@ -154,6 +154,69 @@ pub unsafe extern "C" fn tc_server_new_remote(
 
 #[ffizz_header::item]
 #[ffizz(order = 802)]
+/// Create a new TCServer that connects to the Google Cloud Platform.  See the TaskChampion docs
+/// for the description of the arguments.
+///
+/// On error, a string is written to the error_out parameter (if it is not NULL) and NULL is
+/// returned.  The caller must free this string.
+///
+/// The server must be freed after it is used - tc_replica_sync does not automatically free it.
+///
+/// ```c
+/// EXTERN_C struct TCServer *tc_server_new_gcp(struct TCString bucket,
+///                                       struct TCString credentia_path,
+///                                       struct TCString encryption_secret,
+///                                       struct TCString *error_out);
+/// ```
+#[no_mangle]
+pub unsafe extern "C" fn tc_server_new_gcp(
+    bucket: TCString,
+    credential_path_argument: TCString,
+    encryption_secret: TCString,
+    error_out: *mut TCString,
+) -> *mut TCServer {
+    wrap(
+        || {
+            // SAFETY:
+            //  - bucket is valid (promised by caller)
+            //  - bucket ownership is transferred to this function
+            let bucket = unsafe { TCString::val_from_arg(bucket) }.into_string()?;
+
+            // SAFETY:
+            //  - credential_path is valid (promised by caller)
+            //  - credential_path ownership is transferred to this function
+
+            let credential_path: String;
+
+            if credential_path_argument.is_null() {
+                credential_path = String::from("");
+            } else {
+                credential_path =
+                    unsafe { TCString::val_from_arg(credential_path_argument) }.into_string()?;
+            }
```

Comment:
I talked about `is_null` earlier but I think this isn't actually what we want: TCStrings aren't actually allowed to be null. The `is_null` is a bit of a hack to handle some corner cases where there's a failure converting the TCString from one variant to the other. Specifically, there's no `tc_string_null` to create a null TCString from C.

Instead, I think we'll need to convert this to a `String`, and then check whether that is `""`.
```suggestion
            let credential_path =
                    unsafe { TCString::val_from_arg(credential_path_argument) }.into_string()?;
            let credential_path = if credential_path == "" {
                None
            } else {
                Some(credential_path)
            };
```

Reply:
Okay!